airflow-core/src/airflow/executors/local_executor.py

```diff
@@ -186,8 +186,7 @@ def _check_workers(self):
         # If we're using spawn in multiprocessing (default on macOS now) to start tasks, this can get called a
         # via `sync()` a few times before the spawned process actually starts picking up messages. Try not to
         # create too much
-        need_more_workers = len(self.workers) < num_outstanding
-        if need_more_workers and (self.parallelism == 0 or len(self.workers) < self.parallelism):
+        if num_outstanding and (self.parallelism == 0 or len(self.workers) < self.parallelism):
```

Comment:
I added test cases in https://github.com/aws-mwaa/upstream-to-airflow/pull/15  -> adding a comment here so that this PR is not merged without them,